### PR TITLE
PRC-71 : Complete name validation

### PR DESCRIPTION
**Pull request by james-baker-moj**
Added special char and number validation on names with a dedicated validation test for enter name.

### CODE REVIEW

File: server/routes/contacts/create/enter-name/createContactEnterNameSchemas.test.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        }
+
+        // When
+        const result = await doValidate(form)
+
+        // Then
+        expect(result.success).toStrictEqual(true)
+      },
+    )
+
```
Comment (timharrison-moj) [inline server/routes/contacts/create/enter-name/createContactEnterNameSchemas.test.ts:130]: This is decent validation schema test - which I can `borrow` ...

File: server/routes/contacts/create/enter-name/createContactEnterNameSchemas.ts [hunk 2]
```diff
@@ -3,11 +3,16 @@ import { createSchema } from '../../../../middleware/validationMiddleware'
...
+const LAST_NAME_INVALID = "Contact's last name must not contain special characters"
 
 const FIRST_NAME_REQUIRED_MESSAGE = "Enter the contact's first name"
 const FIRST_NAME_TOO_LONG_ERROR_MSG = "Contact's first name must be 35 characters or less"
+const FIRST_NAME_INVALID = "Contact's first name must not contain special characters"
 
 const MIDDLE_NAME_TOO_LONG_ERROR_MSG = "Contact's middle name must be 35 characters or less"
+const MIDDLE_NAME_INVALID = "Contact's middle name must not contain special characters"
+
+const NAME_REGEX = /^[a-zA-Z\s,.'-]*$/
```
Comment (timharrison-moj) [inline server/routes/contacts/create/enter-name/createContactEnterNameSchemas.ts:15]: Will be fine. Unless one of Frank Zappa's offspring become a contact.

**Review by timharrison-moj — APPROVED** [review]
LGTM

File: server/routes/contacts/create/enter-name/createContactEnterNameSchemas.ts (hunk 2 above)
Comment (james-baker-moj) [reply server/routes/contacts/create/enter-name/createContactEnterNameSchemas.ts:15]: Yeah, I would prefer nothing at all TBH but did a bit of googling and the general consensus is that this accounts for most things.

**Pull request merged**